pkg/apis/pipeline/v1/param_types.go

```diff
@@ -325,11 +325,26 @@ func validatePipelineParametersVariablesInMatrixParameters(matrix []Param, prefi
 	return errs
 }
 
-func validateParametersInTaskMatrix(matrix *Matrix) (errs *apis.FieldError) {
+// validateParamTypesInMatrix validates the type of parameter
+// for Matrix.Params and Matrix.Include.Params
+// Matrix.Params must be of type array. Matrix.Include.Params must be of type string.
+func validateParamTypesInMatrix(matrix *Matrix) (errs *apis.FieldError) {
 	if matrix != nil {
-		for _, param := range matrix.Params {
-			if param.Value.Type != ParamTypeArray {
-				errs = errs.Also(apis.ErrInvalidValue("parameters of type array only are allowed in matrix", "").ViaFieldKey("matrix", param.Name))
+		if matrix.MatrixHasInclude() {
+			for _, include := range matrix.Include {
+				for _, param := range include.Params {
+					if param.Value.Type != ParamTypeString {
+						errs = errs.Also(apis.ErrInvalidValue("parameters of type string only are allowed in matrix", "").ViaFieldKey("matrix.include.params", param.Name))
```

Comment:
only applies to matrix.include.params, not all of matrix

```suggestion
						errs = errs.Also(apis.ErrInvalidValue("parameters of type string only are allowed", "").ViaFieldKey("matrix.include.params", param.Name))
```